File: DPmenu/paiza7.py
n, x, a, y, b = map(int, input().split())
dp = [10000000] * (n + y)
dp[0] = 0

for i in range(x, n + y):
    if i - x >= 0:
        dp[i] = min(dp[i], dp[i - x] + a)
    if i - y >= 0:
        dp[i] = min(dp[i], dp[i - y] + b)

print(min(dp[n:]))

File: Brank2/paiza7.swift
let input = readLine()!.split(separator: " ").map { Int($0)! }
let n = input[0]
let x = input[1]
let k = input[2]
var totalLeftCount = 0
// トータルで左端を通る回数
if k % 4 == 0 {
     totalLeftCount = k / 2
} else {
     totalLeftCount = (k - 1) / 2
}
//Xcm遠ざける前に左端を通る回数。
let leftCount = (4 * n) / 2

//Xcm遠ざけた後に通る回数
let longLeftCount = totalLeftCount - leftCount 

let extraDistance = longLeftCount * x 
print(extraDistance)
